### PR TITLE
DL: Update madlib_keras_fit code to weight by images instead of buffers

### PR DESCRIPTION
**Pull request by khannaekta**
JIRA: MADLIB-1310

Previously, while merging the weights, accuracy and loss, we
assume that all the buffers are of the same size. This may not be always
the case.  Ideally we should use the total no of images per segment to
average the weights, accuracy and loss.
This commit, updates the transition, merge and final function for
`madlib_keras_fit()` to use the number of images on a segment instead of
the buffer count for averaging weights, accuracy and loss.

Additionally, the function `deserialize_weights()` when deserializing
loss and accuracy from state, type-casted it to an int, which resulted
in computing lower loss and accuracy while training. This PR also
addresses this issue.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/853/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/854/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/855/

File: src/ports/postgres/modules/deep_learning/madlib_keras.py_in [hunk 1]
```diff
@@ -82,12 +89,25 @@ def fit(schema_madlib, source_table, model, dependent_varname,
...
         gp_segment_id_col =  -1
     else:
+        # Compute total images on each segment
+        total_images_per_seg = plpy.execute(
+            """ SELECT gp_segment_id, SUM(ARRAY_LENGTH({0}, 1)) AS total_images_per_seg
+                FROM {1}
+                GROUP BY gp_segment_id
+            """.format(dependent_varname, source_table))
+        seg_ids_train = [int(each_segment["gp_segment_id"])
+                   for each_segment in total_images_per_seg]
```
Comment (njayaram2) [inline src/ports/postgres/modules/deep_learning/madlib_keras.py_in]: Might be a good idea to refactor this out to a different function similar to `get_rows_per_seg_from_db`.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/860/

File: src/ports/postgres/modules/deep_learning/madlib_keras.py_in (hunk 1 above)
Comment (reductionista) [reply src/ports/postgres/modules/deep_learning/madlib_keras.py_in]: I went ahead and refactored this, since the same function is needed for the [keras_evaluate() UDA](https://issues.apache.org/jira/browse/MADLIB-1332) I'm working on.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/861/
